runtime/sam/expr/function/time.go

```diff
@@ -21,30 +21,36 @@ type Bucket struct {
 }
 
 func (b *Bucket) Call(_ super.Allocator, args []super.Value) super.Value {
+	args = underAll(args)
 	tsArg := args[0]
 	binArg := args[1]
 	if tsArg.IsNull() || binArg.IsNull() {
+		if tsArg.Type().ID() == super.IDDuration {
+			return super.NullDuration
+		}
 		return super.NullTime
 	}
```

Comment:
Nit: Check argument types before nulls. Doing it this way means `yield bucket(string(null), ip(null))` evaluates to `null(time)`, which is weird.